### PR TITLE
chore(doc): adding volume capacity roundoff details in readme

### PR DESCRIPTION
**Pull request by pawanpraka1**
Signed-off-by: Pawan <pawan@mayadata.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -237,6 +237,55 @@ spec:
...
+spec:
+  storageClassName: openebs-zfspv
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 4G
+```
+
+Here note that storage is 4G. The volume size requested in PVC 4G (4 * 1000 * 1000 * 1024) is not multiple of 4k (4 * 1024, for zfs k is 1024). So, volume creation will fail. We can fix it by changing the storage size to 4Gi which is 4 * 1024 * 1024 * 1024, which is multiple of 4k (4 * 1024).
```
Comment (w3aman) [inline README.md:270]: 4G (4 *1000*1000*1000) **

File: README.md [hunk 2]
```diff
@@ -237,6 +237,55 @@ spec:
...
+metadata:
+  name: openebs-zfspv
+parameters:
+  poolname: "zfspv-pool"
+provisioner: zfs.csi.openebs.io
+```
+
+Here PVC size should be multiple of 8k (8 * 1024).
+
+2. If storageclass has fstype as "zfs" then we don't need to worry about the size. We can use any size we want as in this case a dataset will be created not zvol.
```
Comment (w3aman) [inline README.md:287]: for zfs we can add, that recordsize field should be in power of 2.

Comment (pawanpraka1) [reply README.md:287]: as this section for PVC, added that detail at the other place.

File: README.md [hunk 3]
```diff
@@ -237,6 +237,55 @@ spec:
...
+metadata:
+  name: openebs-zfspv
+parameters:
+  volblocksize: "4k"
+  fstype: "ext4"
+  poolname: "zfspv-pool"
+provisioner: zfs.csi.openebs.io
+```
+
+Here, PVC should have the storage multiple of volblocksize which is 4k (4 * 1024). If is not the multiple of volblocksize then ZFS will not create the volume and pod will not able able to come into RUNNING state as mount will keep on failing. For example, if we have created the below PVC with the above storageclass
```
Comment (w3aman) [inline README.md:254]: small typo ** will not able able ==> will not be able

File: README.md [hunk 4]
```diff
@@ -237,6 +237,55 @@ spec:
...
+metadata:
+  name: openebs-zfspv
+parameters:
+  volblocksize: "4k"
+  fstype: "ext4"
+  poolname: "zfspv-pool"
+provisioner: zfs.csi.openebs.io
+```
+
+Here, PVC should have the storage multiple of volblocksize which is 4k (4 * 1024). If is not the multiple of volblocksize then ZFS will not create the volume and pod will not be able to come into RUNNING state as mount will keep on failing. For example, if we have created the below PVC with the above storageclass
```
Comment (kmova) [inline README.md:254]: @pawanpraka1 - can we someone avoid this manual calculation. When setting the value on the zfs, can we not round it up to the next highest number which  is a multiple of the block/record size?

Comment (pawanpraka1) [reply README.md:254]: that will violate the concept of pvc claim. Provisioned size will be more than claimed size. Provisioned size should be equal to PVC size. Doing round up would not be the right approach IMHO.

Comment (kmova) [reply README.md:254]: Hmm.. so it is about usability vs bytes. another case where this is accepted in practice is when we format a filesystem of 100G, it doesn't always give us 100G usable capacity, so a few bytes is ok in that perspective. 

It is probably harder to do the math that the size should be strictly a multiple of 1024. It has to be rounded off to the closest multiple, I am assuming the diff will be a max of record size == like 4k to 128k bytes difference. 

It is very common for users to request for size as: 
- 5G
- 5Gi

If either of them fails, it violates the ease of use principle.

Comment (pawanpraka1) [reply README.md:254]: @kmova I somehow don't feel that is the right thing. are we doing the same thing in cstor? How are we handeling it there?

Comment (pawanpraka1) [reply README.md:254]: as per discussion rounding off the size seems to be good.

Comment (kmova) [inline README.md:254]: This one needs an update based on the implementation change. https://github.com/openebs/zfs-localpv/pull/191

Comment (pawanpraka1) [reply README.md]: updated @kmova

File: README.md [hunk 5]
```diff
@@ -237,6 +237,56 @@ spec:
 
 Create a PVC using the storage class created for the ZFS driver.
 
+Here, we have to note that the size will be rounded off to the nearest Mi or Gi unit. Please note that M/G notation uses 1000 base and Mi/Gi notation uses 1024 base, so 1M will be 1000 * 1000 byte and 1Mi will be 1024 * 1024.
```
Comment (kmova) [inline README.md:240]: We can probably push the detailed explanation into FAQ and just have a couple of lines about this in the main README.

**Pull request merged**